Riot/Model/HomeserverConfiguration/HomeserverConfigurationBuilder.swift

```diff
@@ -28,23 +28,36 @@ final class HomeserverConfigurationBuilder: NSObject {
     
     /// Create an `HomeserverConfiguration` from an HS Well-Known when possible otherwise it takes hardcoded values from BuildSettings by default.
     func build(from wellKnown: MXWellKnown?) -> HomeserverConfiguration {
-                
-        let isE2EEByDefaultEnabled: Bool
-        let jitsiPreferredDomain: String
-        
         var vectorWellKnownEncryptionConfiguration: VectorWellKnownEncryptionConfiguration?
         var vectorWellKnownJitsiConfiguration: VectorWellKnownJitsiConfiguration?
         
         if let wellKnown = wellKnown, let vectorWellKnown = self.vectorWellKnownParser.parse(jsonDictionary: wellKnown.jsonDictionary()) {
             vectorWellKnownEncryptionConfiguration = self.getEncryptionConfiguration(from: vectorWellKnown)
             vectorWellKnownJitsiConfiguration = self.getJitsiConfiguration(from: vectorWellKnown)
         }
-        
+
         // Encryption configuration
         // Enable E2EE by default when there is no value
-        isE2EEByDefaultEnabled = vectorWellKnownEncryptionConfiguration?.isE2EEByDefaultEnabled ?? true
+        let isE2EEByDefaultEnabled = vectorWellKnownEncryptionConfiguration?.isE2EEByDefaultEnabled ?? true
+        // Disable mandatory secure backup when there is no value
+        let isSecureBackupRequired = vectorWellKnownEncryptionConfiguration?.isSecureBackupRequired ?? false
+        // Default to `MXKKeyPreSharingWhenTyping` when there is no value
+        let outboundKeysPreSharingMode = vectorWellKnownEncryptionConfiguration?.outboundKeysPreSharingMode ?? .whenTyping
```

Comment:
I wonder do we need to set  a default `.whenTyping` here. A default value already exists in `MXKAppSettings`, this is going to override it as long as some homeserver config exists? I wonder does it make more sense to have `outboundKeysPreSharingMode` as optional and  only update `MXKAppSettings` below if it is non-nil?

Reply:
The homeserverConfiguration object always exists, and it defaults to BuildSettings or other app-defined properties if there is no configuration, so I kept it the same way for added properties (we might want to move every default value to BuildSettings though).  
Maybe it should be named otherwise as this could be misleading. It's more a DefaultsThatAHomeServerCouldOverride but I haven't a good name for it yet 😀 

But we should probably in fact at least avoid doubling on default properties that exist outside of this configuration object. I'll set it as nilable if we keep it available.